onnx/onnx.proto

```diff
@@ -52,17 +52,52 @@ enum Version {
   // version that the  graph is generated from. This helps us set up version
   // control. We should use version as
   //     xx(major) - xx(minor) - xxxx(bugfix)
-  // and we are starting with 00000001.
-  IR_VERSION = 00000001;
+  // and we are starting with 00000001, which was the
+  //  version we published on Oct 10, 2017.
+  IR_VERSION_2017_10_10 = 0x00000001;
+
+  // IR_VERSION 1.0.1 (this proto file) contains the following IR format changes
+  // - Added type discriminator to AttributeProto to support proto3 users
+
+  IR_VERSION = 0x00010001;
 }
 
+
+
 // A named attribute containing either singular float, integer, string
 // and tensor values, or repeated float, integer, string and tensor values.
 // An AttributeProto MUST contain the name field, and *only one* of the
 // following content fields, effectively enforcing a C/C++ union equivalent.
 message AttributeProto {
+
+  // Note: this enum is structurally identical to the OpSchema::AttrType
+  // enum defined in schema.h.  If you rev one, you likely need to rev the other.
+  enum AttributeType {
+    FLOAT = 0;
```

Comment:
Should we add an "Undefined" = 0 as the default one here?

Reply:
Edward, until we decide to switch off of semver, let's not keep this PR out of master for that reason. I hate having hanging PRs we need to keep dealing with merges on.

Reply:
@linkerzhang, We could, but why?

Reply:
If not, for the models generated with old version will always have "float" as attribute type by default, which may be not true.

Reply:
Good catch.  Let me write down the versioning spec (I'm going to write the IR rules first since we are close - op versioning needs more bake time).

For all of our enums, we will need to steal 0 as undefined, which is a protobuf best practice.  @dzhulgakov , when I make this change, do you want me to add an undefined enum in the corresponding C++ enum?